### ISSUE TITLE
Unable to cancel orphaned Job. The DEFAULT JOB TIMEOUT setting is not honored.

### ISSUE THREAD

**Issue by cesan3** [ansible/awx#1840]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
 - API
 - UI

##### SUMMARY
Launched a Job. The job got orphaned after running for several hours.  The job never got canceled even if it reached the DEFAULT JOB TIMEOUT, configured as 27000 (7 hours)

##### ENVIRONMENT
* AWX version: 1.0.6.3
* AWX install method: kubernetes
* Ansible version:  2.5.2
* Operating System: Linux
* Web Browser: Chrome / Firefox

##### STEPS TO REPRODUCE

Run a playbook where one of the tasks hangs there forever. 

##### EXPECTED RESULTS
Expect the timeout set DEFAULT JOB TIMEOUT when is reached, the system cancels or terminates the job.

##### ACTUAL RESULTS
No timeout was honored.

DEFAULT_JOB_TIMEOUT 27000

tower-cli result:
tower-cli job list --status running
3520          441 2018-05-01T19:18:31.102280Z running 82742.098922

**Comment by ryanpetrello** [comment]
> Run a playbook where one of the tasks hangs there forever.

@cesan3 do you have an example of such a task?

**Comment by ryanpetrello** [comment]
I'm having trouble reproducing this:

```shell
curl -sk "https://ryan:ryan@tower.example.org/api/v2/settings/jobs/" | python -m json.tool | grep JOB_TIMEOUT
    "DEFAULT_JOB_TIMEOUT": 10,
```

I ran this playbook, and it timed out:

```yaml
---

- name: 'Test playbook to sleep for a specified interval'
  hosts: all
  gather_facts: false
  vars:
    sleep_interval: 120

  tasks:
    - name: sleep for a specified interval
      command: sleep '{{ sleep_interval }}'
```

![screen shot 2018-05-17 at 10 38 21 am](https://user-images.githubusercontent.com/214912/40184497-7162a2e6-59be-11e8-8cd8-6751c1a32114.png)

**Comment by cesan3** [comment]
The task that failed in my case was a yum operation, on an EL6 server.

The blocking process was subscription-manager, so, yum, when loads the plugin and tries to connect to the subscription server hanged.

I had to bounce rhnsd and get rid of the certd.pid to ensure the process respond.

Why the timeout was not respected is unknown.

**Comment by ryanpetrello** [comment]
@cesan3 I've got a hunch about what's causing this - I think it's a bug in an upstream library awx uses, `pexpect`:

https://github.com/pexpect/pexpect/issues/491

Specifically, I think this change is responsible for the issue you've reported:
https://github.com/ansible/awx/commit/f0d3713e99bbcb60805b03100108cc65c696e2f8

**Comment by ryanpetrello** [comment]
@cesan3 we've just released 1.0.7, which we believe resolves the underlying issue here. You can try it out here: https://github.com/ansible/awx/releases/tag/1.0.7

Let us know if you're still seeing this issue after installing the latest awx - thanks!

**Comment by cesan3** [comment]
Hello @ryanpetrello.  The 1.0.7 release is working nicely. No problems so far.
Thanks,

-Cesar
